extensions/Worker.Extensions.Rpc/src/ConfigurationExtensions.cs

```diff
@@ -13,15 +13,28 @@ internal static class ConfigurationExtensions
         /// </summary>
         /// <param name="configuration">The configuration to get the host URI from.</param>
         /// <returns>The URI representing the functions host.</returns>
+        /// <exception cref="UriFormatException">If <paramref name="configuration"/> contain a value for 'functions-uri', but is not a valid URL.</exception>
         /// <exception cref="InvalidOperationException">
         /// If <paramref name="configuration"/> does not contain the required values.
         /// </exception>
         public static Uri GetFunctionsHostGrpcUri(this IConfiguration configuration)
         {
-            string uriString = $"http://{configuration["HOST"]}:{configuration["PORT"]}";
-            if (!Uri.TryCreate(uriString, UriKind.Absolute, out Uri? grpcUri))
+            Uri? grpcUri;
+            var functionsUri = configuration["functions-uri"];
+            if (functionsUri is not null)
             {
-                throw new InvalidOperationException($"The gRPC channel URI '{uriString}' could not be parsed.");
+                if (!Uri.TryCreate(functionsUri, UriKind.Absolute, out grpcUri))
+                {
+                    throw new UriFormatException($"'{functionsUri}' is not a valid value for 'functions-uri'. Value should be a valid URL.");
```

Comment:
If we do map the switch to another configuration name, this validation would need to occur in a different location, as this code wouldn't know anything about the switch name. We need to use the configuration key here instead.